### PR TITLE
Add ClientModule deprecation nagging

### PR DESCRIPTION
**Pull request by jvandort**
Fixes https://github.com/gradle/gradle/issues/25403

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by jvandort** [comment]
@bot-gradle test ast

**Comment by jvandort** [comment]
@bot-gradle test ast

**Comment by jvandort** [comment]
@bot-gradle test ast

**Comment by jvandort** [comment]
@bot-gradle test ast

File: platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DefaultDependencyHandler.java [hunk 1]
```diff
@@ -244,6 +245,13 @@ public Dependency project(Map<String, ?> notation) {
     @Override
     @SuppressWarnings("rawtypes")
     public Dependency module(Object notation, @Nullable Closure configureClosure) {
+
+        DeprecationLogger.deprecate("Declaring client module dependencies")
```
Comment (octylFractal) [inline platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DefaultDependencyHandler.java:249]: Can a more specific deprecation be used here, such as `deprecateAction`?

Comment (jvandort) [reply platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DefaultDependencyHandler.java:249]: I changed it to `deprecateAction` but that did not affect the deprecation message at all. Is there something wrong with the `.deprecate()` message that is printed already?

Comment (jvandort) [reply platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DefaultDependencyHandler.java:249]: It formats the advice if you use `replaceWith`. Updated to use `deprecateAction`

**Review by octylFractal — APPROVED** [review]

**Comment by jvandort** [comment]
@bot-gradle merge

**Comment by jvandort** [comment]
@bot-gradle merge

**Pull request merged**